sql/core/src/main/scala/org/apache/spark/sql/internal/BaseSessionStateBuilder.scala

```diff
@@ -92,15 +82,15 @@ abstract class BaseSessionStateBuilder(
     parentState.map { s =>
       val cloned = s.conf.clone()
       if (session.sparkContext.conf.get(StaticSQLConf.SQL_LEGACY_SESSION_INIT_WITH_DEFAULTS)) {
-        mergeSparkConf(cloned, session.sparkContext.conf)
+        SQLConf.mergeSparkConf(cloned, session.sparkContext.conf)
       }
       cloned
     }.getOrElse {
       val conf = new SQLConf
-      mergeSparkConf(conf, session.sparkContext.conf)
-      options.foreach {
-        case (k, v) => conf.setConfString(k, v)
-      }
+      SQLConf.mergeSparkConf(conf, session.sharedState.conf)
+      // the later added configs to spark conf shall be respected too
+      SQLConf.mergeNonStaticSQLConfigs(conf, session.sparkContext.conf.getAll.toMap)
+      SQLConf.mergeNonStaticSQLConfigs(conf, session.initialSessionOptions)
```

Comment:
why we don't look at `options` now?

Reply:
it is same as session.initialSessionOptions